core/src/main/java/com/microsoft/applicationinsights/internal/channel/common/TelemetryBuffer.java

```diff
@@ -116,17 +116,21 @@ public TelemetryBuffer(TelemetriesTransmitter sender, int maxTelemetriesInBatch,
      * Sets the maximum number of telemetries in a batch
      * @param maxTelemetriesInBatch The max amount of Telemetries that are allowed in a batch.
      */
-    public void setMaxTelemetriesInBatch(int maxTelemetriesInBatch)
-    {
-        this.maxTelemetriesInBatch = maxTelemetriesInBatch;
+    public void setMaxTelemetriesInBatch(int maxTelemetriesInBatch) {
+        synchronized (lock) {
+            if (telemetries != null && maxTelemetriesInBatch < telemetries.size()) {
+                // Request for smaller buffers, we flush if our buffer contains more elements
+                flush();
```

Comment:
Is flush() public? If not we might want it to be so we can use it in other places as well.